system/lib/wasmfs/file.h

```diff
@@ -87,6 +89,8 @@ class File : public std::enable_shared_from_this<File> {
     // specified by the parent weak_ptr.
     std::shared_ptr<File> getParent() { return file->parent.lock(); }
     void setParent(std::shared_ptr<File> parent) { file->parent = parent; }
+
+    backend_t getBackend() { return file->backend; }
```

Comment:
Accessing this read-only field doesn't need to acquire the lock AFAICT, so it could be a method on the `File` rather than the `File::Handle`.